inst/rmd/lua/anchor-sections.lua

```diff
@@ -0,0 +1,17 @@
+--[[
+anchor-sections - Add anchors to headings
+
+# MIT License
+
+Copyright (c) 2020 Atsushi Yasumoto
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+]]
+
+function Header(el)
+  table.insert(el.content, pandoc.RawInline('html',
+    '<a href="#'..el.identifier..'" class="anchor"></a>'))
+  return(el)
+end
```

Comment:
Let's note that if `auto_identifier` pandoc extension is deactivated there won't be any id. (it will be empty `""`)
Note sure anyone will do that, but it is easy to add a check here if we think it is important.

Also, what about using the class `hasAnchor` instead of `anchor` as in **pkgdown** ?

Reply:
> Also, what about using the class hasAnchor instead of anchor as in pkgdown ?

Oh sorry I am misleading here. `hasAnchor` is added to the header tag not the `<a>` tag.

This is useful for some customisation it seems, like hidding the anchor icon on small screen. 
See https://desiree.rbind.io/post/2019/bookdown-anchor-links/

Reply:
I have just seen you simplified it and remove the hasAnchor (https://github.com/rstudio/rmarkdown/pull/1884#issuecomment-683486932)

I guess both way are ok. So it may be fine like this I guess... 🤷‍♂️

Reply:
Thanks. I added the check of identifiers via 295cddfd.
`hasAnchor` is not necessary as we can control visibility for example by adding

```
@media (min-width: 1000px) {
  h1:hover a.anchor::before {
    visibility: hidden
  }
}
```